src/adapters/poi/bragi_poi.js

```diff
@@ -18,106 +18,64 @@ function roundWithPrecision(value, precision) {
 
 export default class BragiPoi extends Poi {
   constructor(feature, queryContext) {
+    const geocodingProps = feature.properties.geocoding;
+    const { id, type, label, address, city, administrative_regions } = geocodingProps;
 
     let poiClassText = '';
     let poiSubclassText = '';
 
-    if (feature.properties.geocoding.properties &&
-        feature.properties.geocoding.properties.length > 0) {
-      const poiClass = feature.properties.geocoding.properties.find(property => {
-        return property.key === 'poi_class';
-      });
-
+    if (geocodingProps.properties && geocodingProps.properties.length > 0) {
+      const poiClass = geocodingProps.properties.find(property => property.key === 'poi_class');
       if (poiClass) {
         poiClassText = poiClass.value;
       }
-      const poiSubclass = feature.properties.geocoding.properties.find(property => {
-        return property.key === 'poi_subclass';
-      });
+      const poiSubclass = geocodingProps.properties.find(property =>
+        property.key === 'poi_subclass');
       if (poiSubclass) {
         poiSubclassText = poiSubclass.value;
       }
     }
-    let addressLabel = '';
-    if (feature.properties &&
-        feature.properties.geocoding &&
-        feature.properties.geocoding.address) {
-      addressLabel = feature.properties.geocoding.address.label;
-    }
 
     /* generate name corresponding to poi type */
     let name = '';
     let alternativeName = '';
-    const adminLabel = '';
-    const resultType = feature.properties.geocoding.type;
 
-    let postcode;
-    if (feature.properties.geocoding.postcode) {
-      postcode = feature.properties.geocoding.postcode.split(';')[0];
-    }
-    const city = feature.properties.geocoding.city;
-    const country = feature.properties.geocoding.administrative_regions.find(administrativeRegion =>
-      administrativeRegion.zone_type === 'country'
-    );
-    const cityObj = feature.properties.geocoding.administrative_regions.find(administrativeRegion =>
-      administrativeRegion.zone_type === 'city'
-    );
-    let countryName;
-    if (country) {
-      countryName = country.name;
-    }
+    const postcode = geocodingProps.postcode && geocodingProps.postcode.split(';')[0];
+    const cityObj = administrative_regions.find(region => region.zone_type === 'city');
+    const country = administrative_regions.find(region => region.zone_type === 'country');
+    const countryName = country && country.name;
 
-    switch (resultType) {
+    switch (type) {
     case 'poi':
-      name = feature.properties.geocoding.name;
-      alternativeName = addressLabel || cityObj && cityObj.label ||
-        [postcode, city, countryName].filter(zone => zone).join(', ');
+      name = geocodingProps.name;
+      alternativeName = address && address.label
+        || cityObj && cityObj.label
```

Comment:
I know we have a lint about this but I really wonder if keeping the parens wouldn't be better in the end... The formatting you used makes it easier to read (and thanks for that) but parens would help me read the code. What do you think about it?

Reply:
I'm not a fan either of this rule, but moreover I'd like to avoid this kind of big boolean expression :) On the next pass I'll try to simplify these rules when parsing Idunn and Bragi POIs, so probably it won't stay long.
For the linting rule, don't hesitate to remove it. Or maybe it can be only relaxed https://eslint.org/docs/rules/no-extra-parens#nestedbinaryexpressions

Reply:
I'll send a PR once yours is merged then.